plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/LivenessProbe.java

```diff
@@ -0,0 +1,104 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.core.server.debug;
+
+import java.util.Collection;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.osgi.util.NLS;
+import org.jboss.tools.openshift.core.server.OpenShiftServerUtils;
+
+import com.openshift.restclient.model.IContainer;
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.probe.IProbe;
+
+/**
+ * A class that allows to manipulate an existing liveness probe for a given deployment config.
+ * 
+ * @author Andre Dietisheim
+ * 
+ * @see <a href="https://kubernetes.io/docs/api-reference/v1.8/#probe-v1-core"/>
+ */
+public class LivenessProbe {
+
+	public static final int INITIAL_DELAY = 1 * 60 * 60; // 1h
+	
+	private IDeploymentConfig dc;
+
+	LivenessProbe(IDeploymentConfig dc) {
+		this.dc = dc;
+	}
+	
+	/**
+	 * Sets the initial delay of the liveness probe for the given deployment config.
+	 * No action is taken if there's no such probe. Returns {@code true} if the
+	 * probe was modified, {@code false} otherwise.
+	 * 
+	 * @param delay
+	 * @param monitor
+	 * @return
+	 * @throws CoreException 
+	 */
+	public boolean setInitialDelay(DebugContext context, IProgressMonitor monitor) throws CoreException {
+		monitor.subTask(NLS.bind("Setting liveness probe initial delay to {0}", INITIAL_DELAY));
+
+		IProbe probe = getLivenessProbe(dc);
+		if (probe == null) {
+			return false;
+		}
+
+		int initialDelay = probe.getInitialDelaySeconds();
+		if (initialDelay >= INITIAL_DELAY) {
```

Comment:
I don't understand the rationale for this case and why don't you save the original value as an annotation in the deployment config ?

Reply:
@jeffmaury if the existing delay is larger than what we'd want for debugging, why should we change it?